src/server/pkg/watch/op.go

```diff
@@ -0,0 +1,17 @@
+package watch
+
+import etcd "github.com/coreos/etcd/clientv3"
+
+// OpOption is a simple typedef for etcd.OpOption.
+type OpOption etcd.OpOption
+
+// WithPrevKV gets the previous key-value pair before the event happens. If the previous KV is already compacted,
+// nothing will be returned.
+func WithPrevKV() OpOption {
+	return OpOption(etcd.WithPrevKV())
+}
+
+// WithFilterPut discards PUT events from the watcher.
+func WithFilterPut() OpOption {
+	return OpOption(etcd.WithFilterPut())
+}
```

Comment:
Is there a reason to lift these types rather than just using them directly? Seems like this is just some extra typing that's not needed.

Reply:
Right, so I decided to do this to be explicit about which etcd options are supported by our watch package. Basically, these are wrappers for the options that use the type system to reasonably prevent  (you can still bypass it through a cast, but you would not accidentally do that) clients of the watch package from using options that don't play nicely with the functionality we add on top of etcd watchers.